content/upcoming/enry.md

```diff
@@ -0,0 +1,182 @@
+* If there's no language, call to the next strategy.
+* If there are more than one languages, they are appended to a list of candidates and it is given to the next strategy.
+* If there is exactly one language, it is returned as the detected language and the process stops.
+
+When the detection process is falling through the strategies without get a only one language as a outcome, the last strategy `GetLanguagesByClassifier` is obviously reached . This strategy deserves take a look at it.
+
+`GetLanguagesByClassifier` internally uses an object of the type:
+```
+type Classifier interface {
+	Classify(content []byte, candidates map[string]float64) (languages []string)
+}
+```
+
+This interface allow you to implement your own classifier and use it throw the functions:
+```
+func GetLanguageBySpecificClassifier(content []byte, candidates []string, classifier Classifier) (language string, safe bool)
+func GetLanguagesBySpecificClassifier(content []byte, candidates []string, classifier Classifier) (languages []string)
+```
+
+*enry*'s default classifier implementation is a bayesian classifier which follows the same implementation *linguist* uses. It assigns scores to the candidates regarding to its probability, with the highest score assigned to the most likely candidate.
+
+By the way, strategies have a `GetLanguage-` version too that returns only a language, and boolean to indicate the sureness of this result. You can use this functions independently for whatever you want.
+
+If you want to custom *enry*'s default strategies and classifier to use your own implemented strategies and classifiers, you can do it assign them to the variables:
+```
+enry.DefaultStrategies = myStrategies
+enry.DefaultClassifier = myClassifier
+```
+From this point on, `GetLanguage` and `GetLanguages` will use your custom implementation.
+
+## How does enry know all about a lot of languages?
+*enry* uses the information about languages (filenames, interpreters, extensions... associated to each language) that *linguist* keeps on its project.
+
+Specifically, *enry* uses the following files from *linguist*:
+* [languages.yml](https://github.com/github/linguist/blob/master/lib/linguist/languages.yml)
+* [heuristics.rb](https://github.com/github/linguist/blob/master/lib/linguist/heuristics.rb)
+* [vendor.yml](https://github.com/github/linguist/blob/master/lib/linguist/vendor.yml)
+* [documentation.yml](https://github.com/github/linguist/blob/master/lib/linguist/documentation.yml)
+
+This files are parsed to retrieve the necessary information. Then the source files in *enry*  that offer this information to the rest of the project are generated and encapsulated as an internal subpackage [data](https://github.com/src-d/enry/tree/master/data).
+
+Whole process is automatic and you only need to run `go generate` from the project's root directory to launch it. It allows *enry* get updated without complex modifications when *linguist* adds new information.
```

Comment:
s/automatic/automated/
s/It allows *enry* get updated/It allows *enry* to get updated/